src/main/java/net/dv8tion/jda/api/entities/MessageSticker.java

```diff
@@ -0,0 +1,206 @@
+ *
+ * @see Message#getStickers()
+ */
+public class MessageSticker implements ISnowflake
+{
+    private final long id;
+    private final String name;
+    private final String description;
+    private final long packId;
+    private final String asset;
+    private final String previewAsset;
+    private final StickerFormat formatType;
+    private final Set<String> tags;
+
+    /** Template for {@link #getAssetUrl()} */
+    public static final String ASSET_URL = "https://cdn.discordapp.com/stickers/%s/%s.%s";
+
+    public MessageSticker(final long id, final String name, final String description, final long packId, final String asset, final String previewAsset, final StickerFormat formatType, final Set<String> tags)
+    {
+        this.id = id;
+        this.name = name;
+        this.description = description;
+        this.packId = packId;
+        this.asset = asset;
+        this.previewAsset = previewAsset;
+        this.formatType = formatType;
+        this.tags = tags;
+    }
+
+    @Override
+    public long getIdLong()
+    {
+        return id;
+    }
+
+    /**
+     * The name of the sticker.
+     *
+     * @return the name of the sticker
+     */
+    @Nonnull
+    public String getName()
+    {
+        return name;
+    }
+
+    /**
+     * The description of the sticker.
+     *
+     * @return the description of the sticker
+     */
+    @Nonnull
+    public String getDescription()
+    {
+        return description;
+    }
+
+    /**
+     * The ID of the pack the sticker is from.
+     *
+     * @return the ID of the pack the sticker is from
+     */
+    @Nonnull
+    public String getPackId()
+    {
+        return Long.toUnsignedString(getPackIdLong());
+    }
+
+    /**
+     * The ID of the pack the sticker is from.
+     *
+     * @return the ID of the pack the sticker is from
+     */
+    public long getPackIdLong()
+    {
+        return packId;
+    }
+
+    /**
+     * The Discord hash-id of the sticker. This represents the actual asset file in the CDN for the sticker.
+     * <br><b>The URL for fetching sticker assets is currently private.
+     *
+     * @return the Discord hash-id of the sticker
+     */
+    @Nonnull
+    public String getAssetHash()
+    {
+        return asset;
+    }
+
+    /**
+     * The Discord hash-id of the preview of the sticker. This represents the actual preview asset file in the CDN for the sticker.
+     * <br><b>The URL for fetching sticker assets is currently private.
+     *
+     * @return the Discord hash-id of the preview image of the sticker or {@code null} if the sticker has no preview image
+     */
+    @Nullable
+    public String getPreviewAssetHash()
+    {
+        return previewAsset;
+    }
+
+    /**
+     * The url of the sticker asset.
+     *
+     * @return the url of the sticker
+     */
+    @Nonnull
+    public String getAssetUrl()
+    {
+        return String.format(ASSET_URL, id, asset, formatType.getExtension());
```

Comment:
We should probably throw if the format type is unknown

Reply:
We should also make it clear that this asset is not necessarily an image here

Reply:
Throw in the `getExtension` or in this method itself?

Reply:
Probably in this method

Reply:
i think it should rather be in `getExtension` as that now returns `null` which is not really the best design

Reply:
I don't really care either way, but if `getExtension` throws then it needs to be documented on both getters.